### ISSUE TITLE
Utilize new SimpleRobotRulesParser API entry point

### ISSUE THREAD

**Issue by sebastian-nagel** [DigitalPebble/storm-crawler#1086]
With crawler-commons 1.4 (#1085) the robots.txt parser (SimpleRobotRulesParser) provides a new [API entry point to parse the robots.txt content](https://crawler-commons.github.io/crawler-commons/1.4/crawlercommons/robots/SimpleRobotRulesParser.html#parseContent(java.lang.String,byte%5B%5D,java.lang.String,java.util.Collection)):
- it's more efficient by accepting a collection of lower-cased, single-word user-agent product tokens, without the need to tokenize a (comma-separated) list of user-agent strings again with every robots.txt
- user-agent matching is compliant with [RFC 9309 (section 2.2.1)](https://www.rfc-editor.org/rfc/rfc9309.html#name-the-user-agent-line) only if the new API method is used

Using the new API comes with some requirements how the user-agent product tokens are passed:
- as collection with multiple alternative user-agent names as separate entries
- always *lower-cased* because the matching is done on a lower-cased user-agent line
- product tokens should comply with [RFC 9309 (section 2.2.1)](https://www.rfc-editor.org/rfc/rfc9309.html#name-the-user-agent-line)
  - e.g., the default value [Anonymous Coward](https://github.com/DigitalPebble/storm-crawler/blob/f2d29fdb7b8636163d513b81dfa185f509c14649/core/src/main/resources/crawler-default.yaml#L68C38-L68C38) does not. It would be still matched, however
  - while trailing version numbers and additional information in the user-agent line are ignored - e.g., `mybot` would match `MyBot/5.0 (http://mybot.info/)` - it does not the other way around: that is, configuring `http.agent.name` as `Mozilla/5.0 (compatible: MyBot/5.0; http://mybot.info/)` would hardly work. Btw., using the old API method there are also unwanted side effects, see crawler-commons/crawler-commons#192.

In short, some extra checks and lower-casing is required, as well as more documentation (should also add the "expert" properties `http.robots.agents` and `http.agent`).

**Comment by jnioche** [comment]
[RobotRulesParser](https://github.com/DigitalPebble/storm-crawler/blob/master/core/src/main/java/com/digitalpebble/stormcrawler/protocol/RobotRulesParser.java) is where things happen

**Comment by jnioche** [comment]
Improved the documentation
Enforce checks of user agent tokens + lowercase them
`http.robots.agents` can be  either a list or a string containing comma separated values
